librz/parse/parse.c

```diff
@@ -99,12 +99,24 @@ RZ_API bool rz_parse_assemble(RzParse *p, char *data, char *str) {
 }
 
 // data is input disasm, str is output pseudo
-// TODO: refactooring, this should return char * instead
-RZ_API bool rz_parse_parse(RzParse *p, const char *data, char *str) {
-	rz_return_val_if_fail(p && data && str, false);
-	return (p && data && *data && p->cur && p->cur->parse)
-		? p->cur->parse(p, data, str)
-		: false;
+RZ_API char *rz_parse_parse(RzParse *p, const char *data) {
```

Comment:
I suggest to rename the function and give it more meaningful name. Also please add a Doxygen